pkgs/os-specific/linux/zfs/2_1.nix

```diff
@@ -0,0 +1,40 @@
+{ callPackage
+, kernel ? null
+, stdenv
+, linuxKernel
+, removeLinuxDRM ? false
+, lib
+, nixosTests
+, fetchpatch
+, ...
+} @ args:
+
+let
+  stdenv' = if kernel == null then stdenv else kernel.stdenv;
+in
+callPackage ./generic.nix args {
+  # You have to ensure that in `pkgs/top-level/linux-kernels.nix`
+  # this attribute is the correct one for this package.
```

Comment:
There is not a free way to assert this? Might be over engineered.

Reply:
Not really, I leave this open on purpose because you might have a reason to use an invalid attribute locally for testing.